From 7b22fc28aa8d795f4934884f57948f0668f47d2e Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@foss.st.com>
Date: Tue, 4 Jun 2024 15:16:09 +0200
Subject: [PATCH 624/841] feat(fdts): add stm32mp215f-dk system configuration
 files

Add bring up (open) config for the stm32mp215f-dk board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 654325a9ab5037d5647912b9cb2a03e7dddc022e feat(fdts): add stm32mp215f-dk system configuration files]
Change-Id: I3d5f508a541d3164d1bee75d423907f3632618bc
Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/385139
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 fdts/stm32mp215f-dk-ca35tdcid-fw-config.dtsi | 34 ++++++++
 fdts/stm32mp215f-dk-ca35tdcid-rcc.dtsi       | 89 ++++++++++++++++++++
 fdts/stm32mp215f-dk-fw-config.dts            |  1 +
 fdts/stm32mp215f-dk.dts                      |  2 +
 4 files changed, 126 insertions(+)
 create mode 100644 fdts/stm32mp215f-dk-ca35tdcid-fw-config.dtsi
 create mode 100644 fdts/stm32mp215f-dk-ca35tdcid-rcc.dtsi

diff --git a/fdts/stm32mp215f-dk-ca35tdcid-fw-config.dtsi b/fdts/stm32mp215f-dk-ca35tdcid-fw-config.dtsi
new file mode 100644
index 000000000..78aeea0aa
--- /dev/null
+++ b/fdts/stm32mp215f-dk-ca35tdcid-fw-config.dtsi
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ */
+
+/*
+ * stm32mp215f tf-a firmware config
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 6/4/2024 2:38:32 PM
+ */
+
+/ {
+	dtb-registry {
+		soc_fw-config {
+			load-address = <0x0 0x81fc0000>;
+			max-size = <0x40000>;
+		};
+		tos_fw {
+			load-address = <0x0 0x82000000>;
+			max-size = <0x2000000>;
+		};
+	};
+
+	st-mem-firewall {
+		bl31_context: bl31-context@81fc0000 {
+			reg = <0x0 0x81fc0000 0x0 0x40000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		};
+		op_tee: op-tee@82000000 {
+			reg = <0x0 0x82000000 0x0 0x2000000>;
+			st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+		};
+	};
+};
diff --git a/fdts/stm32mp215f-dk-ca35tdcid-rcc.dtsi b/fdts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
new file mode 100644
index 000000000..a12ad264d
--- /dev/null
+++ b/fdts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Loic Pallardy loic.pallardy@foss.st.com for STMicroelectronics.
+ */
+
+/*
+ * stm32mp215f Clock tree device tree configuration
+ * Project : open
+ * Generated by XLmx tool version 2.2 - 6/4/2024 2:38:31 PM
+ */
+
+&clk_hse {
+	clock-frequency = <40000000>;
+};
+
+&clk_hsi {
+	clock-frequency = <64000000>;
+};
+
+&clk_lse {
+	clock-frequency = <32768>;
+};
+
+&clk_lsi {
+	clock-frequency = <32000>;
+};
+
+&clk_msi {
+	clock-frequency = <16000000>;
+};
+
+&rcc {
+	st,busclk = <
+		DIV_CFG(DIV_LSMCU, 1)
+		DIV_CFG(DIV_APB1, 0)
+		DIV_CFG(DIV_APB2, 0)
+		DIV_CFG(DIV_APB3, 0)
+		DIV_CFG(DIV_APB4, 0)
+		DIV_CFG(DIV_APB5, 0)
+		DIV_CFG(DIV_APBDBG, 0)
+	>;
+
+	st,flexgen = <
+		FLEXGEN_CFG(0, XBAR_SRC_PLL4, 0, 3)
+		FLEXGEN_CFG(1, XBAR_SRC_PLL4, 0, 5)
+		FLEXGEN_CFG(2, XBAR_SRC_PLL4, 0, 1)
+		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
+		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
+		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
+		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
+		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
+		FLEXGEN_CFG(58, XBAR_SRC_HSE_KER, 0, 1)
+		FLEXGEN_CFG(63, XBAR_SRC_PLL4, 0, 2)
+	>;
+
+	st,kerclk = <
+		MUX_CFG(MUX_USB2PHY1, MUX_USB2PHY1_FLEX57)
+		MUX_CFG(MUX_USB2PHY2, MUX_USB2PHY2_FLEX58)
+	>;
+
+	pll1: st,pll-1 {
+		st,pll = <&pll1_cfg_1200Mhz>;
+
+		pll1_cfg_1200Mhz: pll1-cfg-1200Mhz {
+			cfg = <30 1 1 1>;
+			src = <MUX_CFG(MUX_MUXSEL5, MUXSEL_HSE)>;
+		};
+	};
+
+	pll2: st,pll-2 {
+		st,pll = <&pll2_cfg_400Mhz>;
+
+		pll2_cfg_400Mhz: pll2-cfg-400Mhz {
+			cfg = <20 1 1 2>;
+			src = <MUX_CFG(MUX_MUXSEL6, MUXSEL_HSE)>;
+		};
+	};
+
+	pll4: st,pll-4 {
+		st,pll = <&pll4_cfg_1200Mhz>;
+
+		pll4_cfg_1200Mhz: pll4-cfg-1200Mhz {
+			cfg = <30 1 1 1>;
+			src = <MUX_CFG(MUX_MUXSEL0, MUXSEL_HSE)>;
+		};
+	};
+};
+
diff --git a/fdts/stm32mp215f-dk-fw-config.dts b/fdts/stm32mp215f-dk-fw-config.dts
index 272c848c4..f02b430e8 100644
--- a/fdts/stm32mp215f-dk-fw-config.dts
+++ b/fdts/stm32mp215f-dk-fw-config.dts
@@ -4,3 +4,4 @@
  */
 
 #include "stm32mp21-fw-config.dtsi"
+#include "stm32mp215f-dk-ca35tdcid-fw-config.dtsi"
\ No newline at end of file
diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index c6b205eee..54ba5b9a6 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -6,11 +6,13 @@
 
 /dts-v1/;
 
+#include <dt-bindings/clock/stm32mp21-clksrc.h>
 #include "stm32mp215.dtsi"
 #include "stm32mp21xf.dtsi"
 #include "stm32mp21-lpddr4-1x16Gbits-1x16bits-800MHz.dtsi"
 #include "stm32mp21-pinctrl.dtsi"
 #include "stm32mp21xxan-pinctrl.dtsi"
+#include "stm32mp215f-dk-ca35tdcid-rcc.dtsi"
 
 / {
 	model = "STMicroelectronics STM32MP215F-DK Discovery Board";
-- 
2.47.2

